src/terms/Hamiltonian.jl

```diff
@@ -39,16 +41,12 @@ end
     kpt = H.kpoint
     nband = size(ψ, 2)
 
-    # Allocate another scratch array:
-    Hψ_fouriers = [similar(Hψ[:, 1]) for tid = 1:Threads.nthreads()]
-
-    # take ψi, IFFT it to ψ_real, apply each term to Hψ_temp and Hψ_real, and add it to Hψi
-    Threads.@threads for iband = 1:nband
-        tid = Threads.threadid()
-        ψ_real = H.scratch.ψ_reals[tid]
-        Hψ_real = H.scratch.Hψ_reals[tid]
-        Hψ_fourier = Hψ_fouriers[tid]
+    Hψ_fourier = similar(Hψ[:, 1])
+    ψ_real = zeros(complex(T), basis.fft_size...)
+    Hψ_real = zeros(complex(T), basis.fft_size...)
 
+    # take ψi, IFFT it to ψ_real, apply each term to Hψ_fourier and Hψ_real, and add it to Hψ
+    for iband = 1:nband
```

Comment:
So no more threading over bands for the fallback implementation?

Reply:
I wanted an implementation as simple as possible, so either no threads or no preallocation. It doesn't matter too much, the optimized implementation covers pretty much everything we're interested in right now; we'll revisit when we try out new things

Reply:
Just wanted to make sure this was intentional. Indeed there are few model which come to my mind which are not covered by the fast implementation and where speed should truly matter.